src/Options/ServerOptions.php

```diff
@@ -92,7 +92,7 @@ public function __construct($options = null)
      *
      * @param  array $options
      */
-    public function setFromArray(array $options)
+    protected function setFromArray(array $options)
```

Comment:
Also, the named constructor should call the private constructor, which sets the properties. This way you always pass through type hints.